### PR TITLE
Add static plugin loading support for HPX plugins

### PR DESCRIPTION
**Pull request by abhishek593**
## Proposed Changes

Plugins (binary filters, message handlers) could not be loaded in statically linked HPX builds. `load_plugins` threw "static loading of plugin not supported". Components already had full static loading infrastructure, but plugins were missing the parallel machinery.

- Added a dedicated static plugin registry (`init_registry_plugin_module`, `init_registry_plugin_factory`, `get_static_plugin_factory`) that mirrors the existing component-side registries but keeps plugin bookkeeping in separate maps.

- Added `load_plugin_factory_static()` which iterates plugin registries and injects `static = 1` into each `[hpx.plugins.*]` ini section. Added `load_plugin_static()` in runtime_support to instantiate plugin factories from the static registry.

- Added unit tests covering the registry infrastructure, `static = 1` injection for single and multi-plugin modules, and the empty-factory edge case.

## Checklist

Not all points below apply to all pull requests.

- [X] I have added a new feature and have added tests to go along with it.
- [ ] I have fixed a bug and have added a regression test.
- [ ] I have added a test using random numbers; I have made sure it uses a seed, and that random numbers generated are valid inputs for the tests.

### CODE REVIEW

**Comment by hkaiser** [comment]
@abhishek593 Thanks for looking into this! Where did you encounter the issue you report?

**Comment by abhishek593** [comment]
@hkaiser This is an existing limitation. There is a FIXME in [runtime_support_server.cpp‎](https://github.com/TheHPXProject/hpx/pull/7180/changes#diff-cdce4bd3e77372ecd8015fe76566530a18d700868e6b1f6a5d50acc373045ef4) which this PR addresses.

**Comment by hkaiser** [comment]
@abhishek593 Please take care of the inspect issues reported:
```
/libs/core/runtime_configuration/tests/unit/static_plugin_registry.cpp: *I* missing #include (cstddef) for symbol std::size_t on line 272
```

File: libs/core/runtime_configuration/include/hpx/runtime_configuration/macros.hpp [hunk 1]
```diff
@@ -141,23 +141,46 @@
...
     HPX_PLUGIN_EXPORT(HPX_PLUGIN_PLUGIN_PREFIX,                                \
         hpx::plugins::plugin_registry_base, PluginType, name, plugin)          \
     /**/
 
+////////////////////////////////////////////////////////////////////////////////
+// Mirror the component module gating pattern (lines 66-93 above):
+//   !APP_NAME && !STATIC_LINKING -> full expansion (dynamic build, library TU)
+//   STATIC_LINKING               -> full expansion (static build, any TU)
+//   APP_NAME only                -> nothing       (dynamic build, exe TU)
+#if !defined(HPX_APPLICATION_NAME) && !defined(HPX_HAVE_STATIC_LINKING)
```
Comment (hkaiser) [inline libs/core/runtime_configuration/include/hpx/runtime_configuration/macros.hpp]: Why is this not enabled for the application executable?

Comment (abhishek593) [reply libs/core/runtime_configuration/include/hpx/runtime_configuration/macros.hpp]: @hkaiser The macro is meant to go in a plugin source file, not in application code. It emits the list of factories that were registered in the same translation unit via HPX_REGISTER_PLUGIN_BASE_REGISTRY, plus a static ctor that hands that list to the runtime. An application TU doesn't register any factories itself, so expanding this there would just produce an empty list function and a no-op ctor. 

Although the previous definition was ungated, this now mirrors the component macros' behaviour.

Comment (hkaiser) [reply libs/core/runtime_configuration/include/hpx/runtime_configuration/macros.hpp]: FWIW, I don't see a reason why we shouldn't allow for a plugin to be implemented in the application itself, especially now that statically linked plugin are being supported.

Comment (hkaiser) [reply libs/core/runtime_configuration/include/hpx/runtime_configuration/macros.hpp]: Also, your test actually doing that.

Comment (abhishek593) [reply libs/core/runtime_configuration/include/hpx/runtime_configuration/macros.hpp]: Yeah, I agree. I was more focused on mirroring the components approach and didn't realize my tests are doing the exact same thing. I have dropped the `HPX_APPLICATION_NAME/HPX_HAVE_STATIC_LINKING` gate on `HPX_REGISTER_PLUGIN_REGISTRY_MODULE` and `HPX_REGISTER_PLUGIN_MODULE` so plugins can live in a shared lib, a static build, or the application exe. 

Also had to drop the matching `HPX_HAVE_STATIC_LINKING` guard in `runtime_configuration::load_modules` so exe-registered plugins are actually picked up by the loader.

Comment (abhishek593) [reply libs/core/runtime_configuration/include/hpx/runtime_configuration/macros.hpp]: As a follow-up, would the same reasoning apply to the component macros, or is in-exe components not applicable there for other reasons?

File: libs/core/runtime_configuration/tests/unit/static_plugin_registry.cpp [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+#include <hpx/config.hpp>
+#include <hpx/modules/datastructures.hpp>
+#include <hpx/modules/errors.hpp>
+#include <hpx/modules/ini.hpp>
+#include <hpx/modules/plugin.hpp>
+#include <hpx/modules/testing.hpp>
+#include <hpx/runtime_configuration/init_ini_data.hpp>
+#include <hpx/runtime_configuration/macros.hpp>
+#include <hpx/runtime_configuration/plugin_registry_base.hpp>
+#include <hpx/runtime_configuration/static_factory_data.hpp>
```
Comment (hkaiser) [inline libs/core/runtime_configuration/tests/unit/static_plugin_registry.cpp]: Please use the generated module headers, otherwise this will potentially break the C++ module build:
```suggestion
#include <hpx/modules/runtime_configuration.hpp>
```

File: libs/core/runtime_configuration/include/hpx/runtime_configuration/macros.hpp (hunk 1 above)
Comment (hkaiser) [reply libs/core/runtime_configuration/include/hpx/runtime_configuration/macros.hpp]: Components can be implemented in the executable as well.

Comment (abhishek593) [reply libs/core/runtime_configuration/include/hpx/runtime_configuration/macros.hpp]: Okay, I will create a separate PR for Components then.

**Review by hkaiser — APPROVED** [review]
LGTM, thanks!
